clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/DestroyTitusJobCompletionHandler.java

```diff
@@ -22,25 +22,31 @@
 import com.netflix.spinnaker.clouddriver.titus.TitusCloudProvider;
 import com.netflix.spinnaker.clouddriver.titus.deploy.actions.DestroyTitusJob;
 import com.netflix.spinnaker.clouddriver.titus.deploy.description.DestroyTitusJobDescription;
+import java.util.Optional;
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import org.springframework.stereotype.Component;
 
 @Component
 public class DestroyTitusJobCompletionHandler
-    implements SagaCompletionHandler<DeleteServerGroupEvent> {
+    implements SagaCompletionHandler<Optional<DeleteServerGroupEvent>> {
 
   @Nullable
   @Override
-  public DeleteServerGroupEvent handle(@Nonnull Saga completedSaga) {
+  public Optional<DeleteServerGroupEvent> handle(@Nonnull Saga completedSaga) {
```

Comment:
```
  private fun <T> invokeCompletionHandler(saga: Saga, flow: SagaFlow): T? {
    return flow.completionHandler
      ?.let { completionHandler ->
        val handler = applicationContext.getBean(completionHandler)
        val result = sagaRepository.get(saga.name, saga.id)
          ?.let { handler.handle(it) }
          ?: throw SagaNotFoundException("Could not find Saga to complete by ${saga.name}/${saga.id}")
```

The `SagaService` will raise `SagaNotFoundException` if a completion handler returns a null object.

That does seem suspect (and easily fixed) but it's also nice to indicate when a result object is optional.

Reply:
🤔  Definitely a bug.